AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt

```diff
@@ -812,16 +817,14 @@ open class CardBrowser :
             title = getColUnsafe.undoLabel()
         }
 
-        // Maybe we were called from ACTION_PROCESS_TEXT.
-        // In that case we already fill in the search.
-        if (Intent.ACTION_PROCESS_TEXT == intent.action) {
-            val search = intent.getCharSequenceExtra(Intent.EXTRA_PROCESS_TEXT)
-            if (!search.isNullOrEmpty()) {
-                Timber.i("CardBrowser :: Called with search intent: %s", search.toString())
-                mSearchView!!.setQuery(search, true)
-                intent.action = Intent.ACTION_DEFAULT
-            }
+        launchOptions?.let { options ->
+            if (options !is CardBrowserLaunchOptions.SystemContextMenu) return@let
+            // Fill in the search.
+            Timber.i("CardBrowser :: Called with search intent: %s", launchOptions.toString())
+            mSearchView!!.setQuery(options.search, true)
+            launchOptions = null
```

Comment:
collided with https://github.com/ankidroid/Anki-Android/commit/02ec4291922ab29632853760d762635398a637fb - I think this is the proper way to mash them together?

```suggestion
        launchOptions?.let { options ->
            if (options !is CardBrowserLaunchOptions.SystemContextMenu) return@let
            // Fill in the search.
            Timber.i("CardBrowser :: Called with search intent: %s", launchOptions.toString())
            searchWithFilterQuery(options.search.toString())
            launchOptions = null
        }```

Reply:
It is - pushed